### PR TITLE
Add ignore specific plugins command feature

### PR DESCRIPTION
**Pull request by LordWingZero**
]https://github.com/Tennu/tennu/issues/60#issuecomment-155360581

I have added the ability to pass an array to command-ignore-list, the array must have > 1 strings in it.

The first string is the command, and all subsequent strings are plugins that will have their attempts to register the command with tennu ignored.

**To disable tennu-control say for example:**
```javascript
"command-ignore-list": [ [ "say", "control" ] ]
```

### CODE REVIEW

File: src/plugin/commands.sjs [hunk 1]
```diff
@@ -111,9 +117,31 @@ module.exports = {
...
+                        
+                        if (globalIgnoreItems.indexOf(commandName) !== -1) {
+                            client.note("PluginCommands", format("Ignoring '%s:%s (global)'.", plugin, commandName));
+                            return;
+                        }
+                        
+                        for(var i = 0; i < ignoreCommandInSpecificPluginList.length; i++)
+                        {
+                            var specificIgnore = ignoreCommandInSpecificPluginList[i];
+                            if(specificIgnore.length < 2)
```
Comment (Havvy) [inline src/plugin/commands.sjs]: This check should be done when reading the configuration file.

File: src/plugin/commands.sjs [hunk 2]
```diff
@@ -111,9 +117,31 @@ module.exports = {
...
+                            if(specificIgnore[0] === commandName && specificIgnore.slice(1, specificIgnore.length).indexOf(plugin) !== -1)
+                            {
+                                client.note("PluginCommands", format("Ignoring '%s:%s'.", plugin, commandName));
+                                return;
+                            }
+                        }
+                                            
                         if (commandName in registry) {
-                            const errorMessage = format("Command '%s' already has a handler from plugin '%s'.", commandName, registry[commandName].plugin);
+                            var errorMessage = format("Command '%s' already has a handler from plugin '%s'.", commandName, registry[commandName].plugin);
```
Comment (Havvy) [inline src/plugin/commands.sjs]: This shouldn't have changed.

File: src/plugin/commands.sjs [hunk 3]
```diff
@@ -111,9 +117,31 @@ module.exports = {
...
                             plugin = "UnknownSource";
                         }
-
+                        
+                        if (globalIgnoreItems.indexOf(commandName) !== -1) {
+                            client.note("PluginCommands", format("Ignoring '%s:%s (global)'.", plugin, commandName));
+                            return;
+                        }
+                        
+                        for(var i = 0; i < ignoreCommandInSpecificPluginList.length; i++)
```
Comment (Havvy) [inline src/plugin/commands.sjs]: You want `ignoreCommandInSpecificPluginList.some(...), not a for loop. For loops are against the style of the codebase.

**Comment by LordWingZero** [comment]
Okay, I have fixed the things you have noted. I assume when you said "This check should be done when reading the configuration file." you meant the location of "/src/lib/client.js" because that is where the first read of the config happens.

**Pull request merged**